File: src/main/scala/com/hoi4utils/clausewitz/BoolType.java
package com.hoi4utils.clausewitz;

public enum BoolType {
    TRUE_FALSE("true", "false"),
    YES_NO("yes", "no");

    private final String trueResponse;
    private final String falseResponse;

    BoolType(String trueResponse, String falseResponse) {
        this.trueResponse = trueResponse;
        this.falseResponse = falseResponse;
    }

    public String trueResponse() {
        return trueResponse;
    }

    public String falseResponse() {
        return falseResponse;
    }
    
    public boolean matches(String value) {
        if (value.equalsIgnoreCase(trueResponse)) {
            return true;
        } else if (value.equalsIgnoreCase(falseResponse)) {
            return true;
        }
        return false;
    }
    
    public Boolean parse(String value) {
        if (value.equalsIgnoreCase(trueResponse)) {
            return true;
        } else if (value.equalsIgnoreCase(falseResponse)) {
            return false;
        }
        return null;
    }
}

File: _src_old/main/scala/com/hoi4utils/custom_scala/BoolType.scala
package com.hoi4utils.custom_scala

sealed abstract class BoolType(val trueResponse: String, val falseResponse: String):
  def matches(value: String): Boolean = value.equalsIgnoreCase(trueResponse) || value.equalsIgnoreCase(falseResponse)

  def parse(value: String): Boolean =
    if value.equalsIgnoreCase(trueResponse) then true
    else if value.equalsIgnoreCase(falseResponse) then false
    else null.asInstanceOf[Boolean]


object BoolType:
  case object TRUE_FALSE extends BoolType("true", "false")
  case object YES_NO extends BoolType("yes", "no")

  val values: List[BoolType] = List(TRUE_FALSE, YES_NO)